tests/compile-fail/sync/libc_pthread_rwlock_write_read_deadlock.rs

```diff
@@ -4,10 +4,29 @@
 
 extern crate libc;
 
+use std::cell::UnsafeCell;
+use std::sync::Arc;
+use std::thread;
+
+struct RwLock(UnsafeCell<libc::pthread_rwlock_t>);
+
+unsafe impl Send for RwLock {}
+unsafe impl Sync for RwLock {}
+
+fn new_lock() -> Arc<RwLock> {
+    Arc::new(RwLock(UnsafeCell::new(libc::PTHREAD_RWLOCK_INITIALIZER)))
+}
+
 fn main() {
-    let rw = std::cell::UnsafeCell::new(libc::PTHREAD_RWLOCK_INITIALIZER);
     unsafe {
-        assert_eq!(libc::pthread_rwlock_wrlock(rw.get()), 0);
-        libc::pthread_rwlock_rdlock(rw.get()); //~ ERROR: deadlock
```

Comment:
So the old deadlock detection for a thread waiting on itself does not work any more?
(That's totally okay, I do not think it is very useful, I just want to understand what is happening.)

Reply:
Oh you renamed the test and the diff isn't useful, I see. Never mind.